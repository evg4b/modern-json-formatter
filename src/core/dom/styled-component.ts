import { registerStyles } from '@core/ui/helpers';

export abstract class StyledComponentElement extends HTMLElement {
  protected readonly shadow = this.attachShadow({ mode: 'closed' });
  protected readonly _styles: HTMLStyleElement;

  protected constructor(styles: string) {
    super();
    this._styles = registerStyles(this.shadow, styles);
  }
}
