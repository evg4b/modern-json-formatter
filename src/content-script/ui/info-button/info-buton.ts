import { createElement, CustomElement, StyledComponentElement } from '@core/dom';
import infoButtonStyles from './info-buton.styles';
import icon from './info-button-icon.svg';

@CustomElement('info-button')
export class InfoButtonElement extends StyledComponentElement {
  private readonly link = createElement({
    element: 'a',
    html: icon,
    attributes: {
      target: '_blank',
    },
  });

  constructor(url: string) {
    super(infoButtonStyles);
    this.shadow.appendChild(this.link);
    this.link.href = url;
  }
}
