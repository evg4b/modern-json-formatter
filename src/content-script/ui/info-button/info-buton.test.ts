import '@testing/svg.mock';
import { describe, expect, it } from '@jest/globals';
import { getShadowRoot } from '@testing/styled-component';
import { InfoButtonElement } from './info-buton';

describe('info-button', () => {
  it('should render button link', () => {
    const infoButton = new InfoButtonElement('https://test.com');
    const root = getShadowRoot(infoButton);
    expect(root.innerHTML).toMatchSnapshot();
  });
});
