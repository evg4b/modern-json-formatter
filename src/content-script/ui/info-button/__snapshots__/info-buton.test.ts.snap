// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info-button should render button link 1`] = `"<style type="text/css" rel="stylesheet" role="presentation"></style><a target="_blank" href="https://test.com"><svg>info-button-icon</svg></a>"`;
